client/state/data-layer/wpcom/me/blocks/sites/index.js

```diff
@@ -0,0 +1,46 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { READER_SITE_BLOCKS_REQUEST, READER_SITE_BLOCKS_RECEIVE } from 'state/action-types';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+export const handleSiteBlocksRequest = action =>
+	http(
+		{
+			apiNamespace: 'wpcom/v2',
+			method: 'GET',
+			path: '/me/blocks/sites',
+			query: {
+				page: action.page || null,
+				per_page: action.perPage || null,
+			},
+		},
+		action
+	);
+
+export const siteBlocksRequestReceived = ( action, apiResponse ) => ( {
+	type: READER_SITE_BLOCKS_RECEIVE,
```

Comment:
I'm used to seeing those objects generated from separate action creator functions, mainly for reusability reasons. Is inlining them here done with a specific purpose?

Reply:
if you import the action creators from the data-layer, which is imported by the action creators, you end up with a module dependency loop. Dependency loops can have strange consequences that I'd love to avoid.

Reply:
Main consequence being that the action creators may not be defined if the data layer uses them during module initialization.

Reply:
That makes sense, thanks!